ding/reward_model/trex_reward_model.py

```diff
@@ -0,0 +1,618 @@
+from collections.abc import Iterable
+from easydict import EasyDict
+
+import torch
+import torch.nn as nn
+import torch.nn.functional as F
+import torch.optim as optim
+
+from ding.utils import REWARD_MODEL_REGISTRY
+from .base_reward_model import BaseRewardModel
+from ding.model.template.q_learning import DQN
+from ding.model.template.vac import VAC
+from ding.model.template.qac import QAC
+from torch.distributions.categorical import Categorical
+import gym
+import numpy as np
+from dizoo.atari.envs.atari_wrappers import wrap_deepmind
+from dizoo.mujoco.envs.mujoco_wrappers import wrap_mujoco
+from .rnd_reward_model import collect_states
+from ding.utils import SequenceType
+from ding.model.common import FCEncoder
+from torch.distributions import Normal, Independent
+from ding.utils.data import offline_data_save_type
+from copy import deepcopy
+from ding.utils import build_logger
+
+
+class ConvEncoder(nn.Module):
+    r"""
+    Overview:
+        The ``Convolution Encoder`` used in models. Used to encoder raw 2-dim observation.
+    Interfaces:
+        ``__init__``, ``forward``
+    """
+
+    def __init__(self, ) -> None:
+        super(ConvEncoder, self).__init__()
+        self.conv1 = nn.Conv2d(4, 16, 7, stride=3)
+        self.conv2 = nn.Conv2d(16, 16, 5, stride=2)
+        self.conv3 = nn.Conv2d(16, 16, 3, stride=1)
+        self.conv4 = nn.Conv2d(16, 16, 3, stride=1)
+        self.fc1 = nn.Linear(784, 64)
+        self.fc2 = nn.Linear(64, 1)
+
+    def forward(self, x: torch.Tensor) -> torch.Tensor:
+        x = x.permute(0, 3, 1, 2)  # get into NCHW format
+        # compute forward pass of reward network (we parallelize across frames so
+        # batch size is length of partial trajectory)
+        x = F.leaky_relu(self.conv1(x))
+        x = F.leaky_relu(self.conv2(x))
+        x = F.leaky_relu(self.conv3(x))
+        x = F.leaky_relu(self.conv4(x))
+        x = x.reshape(-1, 784)
```

Comment:
use more general implementation, not hard code

Reply:
Will do this shortly --- combining DI-engine model (encoder + head) to generalise the reward model